Docs/Reference.dox

```diff
@@ -3800,6 +3800,26 @@ byte[]     Bytecode, produced by AngelScript serializer
 
 - Inline functions are defined inside the class definitions where possible, without using the inline keyword.
 
+C++11 code conventions:
+
+- Use \c override wherever possible.
+
+- Prefer inplace member initialization to initializer lists.
+
+- Prefer \c using to \c typedef type declarations.
+
+- Prefer range-based \c for to old style \c for unless index or iterator is used by itself.
+
+- Avoid \c enum \c class to keep Urho API consistent. TODO: This isn't the best decision.
+
+- Avoid \c auto variables unless it increases code readability. For example, \c auto is tolerable for iterators, pairs and long template types.
+  Inspired by https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html
+
+- Prefer \c auto to manual type deduction via \c decltype and \c typename.
+
+- Prefer old style constuctor syntax `Vector3(x, y, z)` to new constructor syntax `Vector3{x, y, z}` unless new syntax increases code readability.
+  Example: `Vector3 vec(x, y, z)` but `node.SetPosition({x, y, z})`
```

Comment:
Would be clearer to say :
Not `node.SetPosition(Vector3(x, y, z))` but `node.SetPosition({x, y, z})`